train_gpt2.cu

```diff
@@ -1636,29 +1504,28 @@ void matmul_backward(floatX* dinp, floatX* dweight, floatX* dbias,
                                        dim3(block_size_x, block_size_y),
                                        OC_per_warp * sizeof(float), main_stream>>>(dbias_buffer, dout, B, T, OC);
         cast_and_add_kernel<<<CEIL_DIV(OC, 256), 256, 0, main_stream>>>(dbias, dbias_buffer, OC);
-        cudaCheck(cudaGetLastError());
```

Comment:
why remove the kernel launch check?

Reply:
I figured it'd be more consistent to always have a single "cudaCheck(cudaGetLastError())" at the end of every function.

That should be more than enough for the initial debug of where the problem comes from without cluttering e.g. attention_forward/backward with 3 error checks each.